state-res: Improve docs in event_auth module and refactor

zecakeh opened the pull request:
The first commit is a simple reorganization of modules.

The second commit is more involved:

- Copy the authorization rules docs from the spec.
- Make sure that the auth rules are applied in the correct order (only necessary for power levels).
- Change variable names to match the docs.
- Add FIXMEs where the implementation deviates from the spec.
- Clean up old docs.

There should be no differences in the output of `auth_check` for now.

zecakeh committed:
state-res: Move state_event and power_levels into a new events module

jplatte commented on crates/ruma-state-res/src/event_auth.rs:
```diff
@@ -753,14 +836,14 @@ fn can_send_event(event: impl Event, ple: Option<impl Event>, user_level: Int) -
-    previous_power_event: Option<impl Event>,
-    user_level: Int,
+    power_level_event: impl Event,
+    previous_power_level_event: Option<impl Event>,
```
Should be called `power_levels_event` (extra `s`) probably?

zecakeh replied:
Yes it should!

I actually fixed that in another branch that will be in a follow-up more invasive PR where I changed the name of the variables again… 

So maybe I should revert those variable name changes here.

zecakeh replied:
Alright I reverted some variable name changes.

jplatte reviewed (approved):
nit: There's an extra "for" in the description of the second commit.

zecakeh committed:
state-res: Improve docs in event_auth module and refactor

- Copy the authorization rules docs from the spec.
- Make sure that the auth rules are applied in the correct order (only
necessary for power levels).
- Change variable names to match the docs.
- Add FIXMEs where the implementation deviates from the spec.
